csm/conf/setup.py

```diff
@@ -142,6 +143,7 @@ def _is_group_exist(user_group):
             return False
 
     @staticmethod
+    @functools.lru_cache(maxsize=None)
```

Comment:
why do we want to set lru cache here? Why it is needed. what if i don't set it, which one may break and in which scenario?

as per my basic understanding of function, this is just good to have and mainly necessary where we want to play with cache a lot. is it the case in this function?

Reply:
`get_data_from_provisioner_cli()` does what the name says, it invokes the provisioner command and grabs its output.

The execution of such command takes seconds.

```
ssc-vm-1154 ~$ time sudo provisioner get_setup_info --out=json
{
    "ret": {
        "nodes": 1,
        "server_type": "virtual",
        "servers_per_node": 1,
        "storage_type": "RBOD"
    }
}

real    0m2.728s
user    0m1.369s
sys     0m0.279s
```

Therefore caching the results is a way to amortize the costs of multiple calls to it.  Of course it assumes that the provisioner will always return the same results for each execution, which must be the case in the uses we have during the setup process---we use it to obtain the setup info and the node ID.